File: 0001-traversing-a-matrix/javascript/main-friendly.js
const MATRIX_SIZE = 10000;
let arr = [];

for(let i = 0; i < MATRIX_SIZE; i++){
    arr[i] = [];
    for(let j = 0; j < MATRIX_SIZE; j++){
        arr[i][j] = Math.random();
    }
}

let result = 0;

beginTime = new Date();
for(let j = 0; j < MATRIX_SIZE; j++){
    for(let i = 0; i < MATRIX_SIZE; i++){
        result += arr[j][i] ;
    }
}
let endTime_row = new Date() - beginTime;

console.log('time by cache friendly: ', endTime_row);
return result;

File: 0001-traversing-a-matrix/java/src/com/weremsoft/MainFriendly.java
package com.weremsoft;

public class MainFriendly {
    public static void main(String[] args) {

        final int MATRIX_SIZE = 10000;

        long result = 0;

        char[][] arrNumbers = new char[MATRIX_SIZE][MATRIX_SIZE];

        long begin = System.currentTimeMillis();
        for (int i = 0; i < MATRIX_SIZE; i++) {
            for (int j = 0; j < MATRIX_SIZE; j++) {
                result += arrNumbers[i][j];
            }
        }
        long end = System.currentTimeMillis();

        double cache_unfriendly = (end - begin);

        System.out.println("Cache Friendly elapsed time: " + cache_unfriendly);
        System.out.println("Result: " + result);
    }
}
